app/src/main/java/ch/epfl/sdp/peakar/map/OSMMap.java

```diff
@@ -297,9 +301,44 @@ public boolean onLongPress(final MotionEvent e, final MapView mapView) {
 
         };
 
-        // TODO add bounding box drawing
         mapView.getOverlays().add(touchOverlay);
-mapView.invalidate();
+        mapView.invalidate();
+
+    }
+
+    /**
+     * Allows to draw a bounding box on the map around the selectedPoint.
+     *
+     * @param selectedPoint         point around which drawing the bounding box.
+     * @param mapView               mapView to draw the boundingBox polygon.
+     * @param boundingBoxPolygon    boundingBoxPolygon already in the overlays.
+     * @return                      the new Polygon that is drawn on the map.
+     */
+    public Polygon drawBoundingBox(Point selectedPoint, MapView mapView, Polygon boundingBoxPolygon) {
+
+        List<GeoPoint> edges = new ArrayList<>();
+        BoundingBox boundingBox = selectedPoint.computeBoundingBox(GeonamesHandler.DEFAULT_RANGE_IN_KM);
+
+        edges.add(new GeoPoint(boundingBox.getLatNorth(), boundingBox.getLonWest()));
+        edges.add(new GeoPoint(boundingBox.getLatNorth(), boundingBox.getLonEast()));
+        edges.add(new GeoPoint(boundingBox.getLatSouth(), boundingBox.getLonEast()));
+        edges.add(new GeoPoint(boundingBox.getLatSouth(), boundingBox.getLonWest()));
+
+        if (boundingBoxPolygon != null) {
+            mapView.getOverlays().remove(boundingBoxPolygon);
+            mapView.invalidate();
+        }
+
+        boundingBoxPolygon = new Polygon();
+        edges.add(edges.get(0));    //forces the loop to close(connect last point to first point)
+        boundingBoxPolygon.getFillPaint().setColor(Color.parseColor("#1EFFE70E")); //set fill color
+        boundingBoxPolygon.setPoints(edges);
```

Comment:
Check method 
`public static ArrayList<IGeoPoint> pointsAsRect(GeoPoint center, double lengthInMeters, double widthInMeters)`
in Polygon. It could simplify the computation here. 
Suggestion : 
`boundingBoxPolygon.setPoints(boundingBoxPolygon.pointsAsRect(selectedPoint,GeonamesHandler.DEFAULT_RANGE_IN_KM,GeonamesHandler.DEFAULT_RANGE_IN_KM))`
Where `selectedPoint` should be converted first as `GeoPoint`

Reply:
The problem is that `pointsAsRect` returns a list of `IGeoPoints`, but a list of `GeoPoints` is required.